utils/logging/logging.go

```diff
@@ -26,8 +47,17 @@ func SetupLogger(ctx context.Context) (context.Context, *zerolog.Logger) {
 		output = zerolog.ConsoleWriter{Out: os.Stdout}
 	}
 
+	// this log writer uses a ring buffer and drops messages that cannot be processed
+	// in a timely manner
+	wr := diode.NewWriter(output, 1000, time.Duration(10*time.Millisecond), func(missed int) {
+		// write to stderr the number of dropped log messages
+		fmt.Fprintf(os.Stderr, "logger dropped message count: %d", missed)
```

Comment:
Should we check the length and only print out if there are any missed log events?

Reply:
This parameter to diode.NewWriter is only called when there are dropped messages.  I think missed will always be > 0 if this function is called.